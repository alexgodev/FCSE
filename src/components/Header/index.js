import React from 'react';
import { Link } from 'react-router-dom';
import { HeaderButton, HeaderWrapper } from './Styled';
import useAuthUser from '../../hooks/useAuthUser';
import { useTranslate } from 'react-translate';

const Header = () => {
    const { isAuth } = useAuthUser();
    const t = useTranslate('Header');

    return (
        <HeaderWrapper>
            <Link to="/">
                <HeaderButton>{t('home')}</HeaderButton>
            </Link>
            {isAuth ? (
                <Link to="profile">
                    <HeaderButton>{t('profile')}</HeaderButton>
                </Link>
            ) : (
                <Link to="login">
                    <HeaderButton>{t('login')}</HeaderButton>
                </Link>
            )}
        </HeaderWrapper>
    );
};

export default Header;
