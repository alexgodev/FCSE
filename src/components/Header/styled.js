import styled from 'styled-components';
import colors from '../../styles/colors';
import { fontWeightExtraBold } from '../../styles/variables';

export const HeaderWrapper = styled.div`
    display: flex;
    justify-content: center;
    width: 100%;
    background-color: ${colors.fillRgba100.blackPrimary};
`;

export const HeaderButton = styled.button`
    color: ${colors.fill.whiteDefault};
    padding: 16px;
    font-weight: ${fontWeightExtraBold};
    text-transform: uppercase;
    letter-spacing: 0.04em;

    &:hover {
        color: ${colors.fill.blueBright};
        background-color: ${colors.fillRgba5.whiteDefault};
    }
`;
