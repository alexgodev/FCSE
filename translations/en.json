{
  "locale": "en",
  "Login": {
    "login": "login",
    "sign_in": "sign in",
    "email_address": "email address",
    "password": "password"
  },
  "Home": {
    "hello_world": "hello world!",
    "try_our_amazing_app": "try our amazing app"
  },
  "Header": {
    "home": "home",
    "profile": "profile",
    "login": "login"
  },
  "Profile": {
    "profile": "profile",
    "first_name": "first name",
    "last_name": "last name",
    "logout": "logout"
  },
  "Error": {
    "an_error_occurred": "an error occurred"
  }
}