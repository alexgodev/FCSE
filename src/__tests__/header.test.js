import React from 'react';
import renderer from 'react-test-renderer';
import { MockedProvider } from '@apollo/client/testing';
import Header from '../components/Header';
import { BrowserRouter } from 'react-router-dom';
import translations from '../../translations/en.json';
import { TranslatorProvider } from 'react-translate';
import { mount } from 'enzyme';

it('render correctly header component', () => {
    const component = renderer.create(
        <BrowserRouter>
            <MockedProvider>
                <TranslatorProvider translations={translations}>
                    <Header />
                </TranslatorProvider>
            </MockedProvider>
            ,
        </BrowserRouter>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('check menu displayed 2 buttons', () => {
    const component = mount(
        <BrowserRouter>
            <MockedProvider>
                <TranslatorProvider translations={translations}>
                    <Header />
                </TranslatorProvider>
            </MockedProvider>
            ,
        </BrowserRouter>,
    );
    expect(component.find('button').length).toEqual(2);
});
