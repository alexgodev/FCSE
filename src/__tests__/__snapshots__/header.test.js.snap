// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly header component 1`] = `
Array [
  <div
    className="sc-bdvvtL clJdMN"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="sc-gsDKAQ gzTrXc"
      >
        home
      </button>
    </a>
    <a
      href="/login"
      onClick={[Function]}
    >
      <button
        className="sc-gsDKAQ gzTrXc"
      >
        login
      </button>
    </a>
  </div>,
  ",",
]
`;
